### DISCUSSION TITLE
Contributing to open source projects - how to start

### DISCUSSION THREAD

**Discussion by marcinjahn** [elementary/edw#13] [Ideas] [answered]
Speaker: **TBD**

Show how to contribute to an open-source project, like elementaryOS. I think it would be really nice to see a practical example of what someone should do if they spot a place for improvement in some part of eOS - like any default app or some system component. Of course we'd assume that the person is a developer. The focus should be on:
1. How to find the source code
2. How to start working on your improvement/fix
3. How to run your own version of some system component to verify the code
4. How to submit the improvement as a pull request

**Comment by cassidyjames** [comment, marked as answer]
I like this, but do think it should be scoped to elementary OS specifically. That's a huge advantage of an open source and openly-developed OS for app developers: they can not only see how the platform works, but very easily contribute to the OS itself.

**Reply by igordsm** [reply]
I could give this talk (if there aren't more core members interested). I teach a course on collaborative development and guide students into their first contributions to FLOSS projects (materials are available in pt_BR: https://insper.github.io/dev-aberto/). I even used elementary in some examples to students. 

One possible idea would be to do a live replay of some merged pull request. The idea would be to address the 4 points of the OP using one of elementary's builtin apps and "redo" a PR from 0. This would include reproducing a bug, downloading and compiling source code, navigating through a completely new code base and finding where things break. I could use as example one of my PRs to [camera](https://github.com/elementary/camera/pulls?q=is%3Apr+is%3Aclosed+author%3Aigordsm) and focus on both code and design discussions.

**Reply by cassidyjames** [reply]
@igordsm that sounds awesome. :smile:

**Reply by igordsm** [reply]
Submitted :)

**Reply by danirabbit** [reply]
@cassidyjames should we consider this closed then? :)
